cmd/authelia-scripts/cmd_docker.go

```diff
@@ -11,7 +11,8 @@ import (
 
 func DockerBuildOfficialImage() error {
 	docker := &Docker{}
-	return docker.Build(IntermediateDockerImageName, ".")
+	Dockerfile := os.Getenv("DOCKERFILE")
```

Comment:
the Dockerfile could be deduced from an -arch flag provided by the user. No need for an env variable.

Reply:
I thought the purpose of these scripts were just to help with your Travis setup.

I didn't really think or know that you intended for users to utilise these too, am I understanding correctly that is what the actual intent is?

Reply:
In my mind the only script not meant to be executed by users is `authelia-scripts ci`. The other ones are here for users to have a built version of Authelia or have a local docker image or run integration tests and unit tests locally. In my mind the interface of all those scripts should be user-friendly and the ci one could be relaxed.

Now I understand the fact that env variables is the way to go in Travis so I guess the best solution would be to define the env variable in Travis and forward them as flag(s) to the CLI. I think the right level of abstraction for that script would be the architecture now. Would it work to have a flag -arch and build the Dockerfile name from that arch? Something like Dockerfile.$arch and then the user (likely me for now) could do `authelia-scripts docker build -arch=amd64`, `authelia-scripts docker push-image -arch=amd64` and `authelia-scripts docker push-manifest`

In the ci script we could probably run something like `authelia-scripts docker build -arch=$ARCH` so that you can export ARCH as the env variable in Travis and make it consumed by the script.

Does it make sense?

Reply:
Per discussion in Gitter I think this will still present issues if a person tries to build different Architectures because of the other steps scattered throughout the different Dockerfiles and `.travis.yml`

I'll update now with a solution so the interface is clear for users and the additional logic is utilised when called via Travis.